### DISCUSSION TITLE
working-on-a-git-branch-in-two-different-folders-concurrently/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#415] [Announcements]
# working-on-a-git-branch-in-two-different-folders-concurrently/

In this post I show how you can work on a git branch with a repository checked out in Windows, and then sync that to another git folder in WSL

https://andrewlock.net/working-on-a-git-branch-in-two-different-folders-concurrently/?ref=dailydev

**Comment by nphamvn** [comment]
I am using rsync (with  .rsyncignore) to sync code between machines.

**Comment by WorldMaker** [comment]
A third option, mentioned also in the error message, is to use a simple bare repo as a go between. `git clone --bare` will create a clone with no working directory (and thus no checked out branch) that both your other clones can use as a remote and push/pull between. The obvious trade off is having a third copy of your repo’s object database on your hard drive (though there are mitigation options there you can explore with care if that’s an issue), but the benefits of having a two-way intermediate clone can be handy for some workflows even if you still want it to be a local bare clone rather than a true centralized “cloud” remote.
